modules/@apostrophecms/piece-widget-type/index.js

```diff
@@ -95,7 +95,7 @@ module.exports = {
 
     if (_.includes(by, 'id')) {
       addFields.push({
-        type: 'joinByArray',
+        type: 'join',
         name: '_pieces',
         label: byChoices.length > 1 ? 'Individually' : 'Select...',
         idsField: 'pieceIds',
```

Comment:
Please remove this, it's legacy bc stuff in 2.x, let it pick what it picks by default (piecesIds). Not your fault, but this is the right time to fix it.